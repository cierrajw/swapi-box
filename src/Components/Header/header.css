@font-face {
  font-family: starWars;
  src: url(../../assets/fonts/Starjhol.ttf)
}

.header-section{
  display: grid;
  background-color: black;
  opacity: .6;
  grid-template-columns: 1.2fr 1fr;
  color: gold;
}

.swapi-title{
  grid-column: 1;
  display: flex;
  justify-self: flex-end;
  font-size: 3em;
}

.favorites-button{
  text-align: center;
  grid-column: 2;
  height: 30px;
  width: 80px;
  display: flex;
  justify-self: flex-end;
  margin: 1em;
}
