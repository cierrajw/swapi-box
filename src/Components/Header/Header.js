import React, {Component} from 'react';
import PropTypes from 'prop-types';
import './header.css'


class Header extends Component{
  constructor(){
    super();
  }

  handleFavoriteClick = () =>{
    this.props.displayFavorites();
  }

  render(){
    const favoriteNumber = this.props.favoriteCards.length;
    return (
      <div className="header-div">
        <section className="header-section">
          <h1 className="swapi-title">SWAPI BOX</h1>
          <button className="favorites-button" onClick={()=>this.handleFavoriteClick()}>Favorites <span>{favoriteNumber}</span></button>
        </section>
      </div>
    );
  }
}

Header.propTypes = {
  favoriteCards: PropTypes.array
}


export default Header;
