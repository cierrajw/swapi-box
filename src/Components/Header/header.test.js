import React from 'react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import Header from './Header.js';

describe('Header', () => {

  it('should match the snapshot', () => {

    const wrapper = shallow(<Header />);

    expect(wrapper).toMatchSnapshot();

  });

});
