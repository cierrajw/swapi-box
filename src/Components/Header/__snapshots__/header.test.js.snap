// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="header-section"
      >
        <h1
          className="swapi-title"
        >
          Swapi Box
        </h1>
        <button
          className="favorites-button"
        >
          Favorites
        </button>
      </section>,
      "className": "header-div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="swapi-title"
          >
            Swapi Box
          </h1>,
          <button
            className="favorites-button"
          >
            Favorites
          </button>,
        ],
        "className": "header-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Swapi Box",
            "className": "swapi-title",
          },
          "ref": null,
          "rendered": "Swapi Box",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorites",
            "className": "favorites-button",
          },
          "ref": null,
          "rendered": "Favorites",
          "type": "button",
        },
      ],
      "type": "section",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="header-section"
        >
          <h1
            className="swapi-title"
          >
            Swapi Box
          </h1>
          <button
            className="favorites-button"
          >
            Favorites
          </button>
        </section>,
        "className": "header-div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="swapi-title"
            >
              Swapi Box
            </h1>,
            <button
              className="favorites-button"
            >
              Favorites
            </button>,
          ],
          "className": "header-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Swapi Box",
              "className": "swapi-title",
            },
            "ref": null,
            "rendered": "Swapi Box",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Favorites",
              "className": "favorites-button",
            },
            "ref": null,
            "rendered": "Favorites",
            "type": "button",
          },
        ],
        "type": "section",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
